### PR TITLE
Chart: Make PSS compliant.

### PR DESCRIPTION
**Pull request by erkanerol**
Towards https://github.com/giantswarm/roadmap/issues/2970

The app is currently failing because of PSSs and it still deploys PSPs even when PSSs are enforced. This PR fixes the issue. Tested in `waper`.

### CODE REVIEW

**Review by weseven — APPROVED** [review]

File: config/helm/patches/deployments/capi-controller-manager.yaml [hunk 1]
```diff
@@ -22,3 +22,6 @@ spec:
         - name: metrics
           protocol: TCP
           containerPort: 8080
+        securityContext:
+          seccompProfile:
+            type: RuntimeDefault
```
Comment (Gacko) [inline config/helm/patches/deployments/capi-controller-manager.yaml:27]: Have you seen this is being deleted a few lines above? Why not just drop the `$patch: delete`? :)

File: helm/cluster-api/templates/crd-install/crd-job.yaml [hunk 2]
```diff
@@ -52,6 +55,13 @@ spec:
           for deployment in $(kubectl get deployment -l app.kubernetes.io/name=cluster-operator -o name); do
             kubectl rollout restart -n giantswarm "$deployment"
           done
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+          seccompProfile:
+            type: RuntimeDefault
```
Comment (Gacko) [inline helm/cluster-api/templates/crd-install/crd-job.yaml:64]: This should be set either on pod or on container level, but it doesn't need to be set on the both of them.

File: helm/cluster-api/templates/hook-delete-aggregated-roles/job.yaml [hunk 3]
```diff
@@ -39,5 +39,16 @@ spec:
...
+          capabilities:
+            drop:
+            - ALL
+          seccompProfile:
+            type: RuntimeDefault
       restartPolicy: Never
+      securityContext:
+        runAsNonRoot: true
+        seccompProfile:
+          type: RuntimeDefault
```
Comment (Gacko) [inline helm/cluster-api/templates/hook-delete-aggregated-roles/job.yaml:53]: Same here. Either container or pod, not both.

**Review by Gacko — CHANGES_REQUESTED** [review]

File: helm/cluster-api/templates/crd-install/crd-psp.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {{- if .Values.crds.install }}
-{{- if .Capabilities.APIVersions.Has "policy/v1beta1/PodSecurityPolicy" }}
+{{- if not .Values.global.podSecurityStandards.enforced }}
```
Comment (Gacko) [inline helm/cluster-api/templates/crd-install/crd-psp.yaml:2]: Please also incorporate this condition in the according role and role binding.

**Review by Gacko — CHANGES_REQUESTED** [review]

**Comment by Gacko** [comment]
Also adding seccomp profile requires adding an annotation to the existing PSP. I'll check if we need that PSP at all.

**Comment by erkanerol** [comment]
> Hope it's fine with you that I just implemented my thoughts and improvements in this PR.

NO. WHY DON'T YOU TOUCH MY PR WITHOUT GETTING MY APPROVAL???????


just kidding :D  I am OK. Please tell me when you are done and let me verify the last version in CAPZ side.

**Comment by erkanerol** [comment]
> Also adding seccomp profile requires adding an annotation to the existing PSP. I'll check if we need that PSP at all.

I haven't hit any issues regarding this on the CAPI side.

**Review by Gacko — APPROVED** [review]

**Pull request merged**